packages/ui-tree-browser/src/TreeBrowser/TreeCollection/props.ts

```diff
@@ -28,40 +28,25 @@ import PropTypes from 'prop-types'
 import { Children } from '@instructure/ui-prop-types'
 
 import { TreeNode } from '../TreeNode'
+import type { CollectionData } from '../props'
 
 import type {
   PropValidators,
   TreeBrowserCollectionTheme
 } from '@instructure/shared-types'
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
+import { CollectionProps, TreeBrowserBaseProps } from '../props'
 
 type TreeBrowserCollectionOwnProps = {
-  id?: string | number
-  name?: string
-  descriptor?: string
-  items?: any[]
-  collections?: any[]
-  expanded?: boolean
-  selection?: string
-  size?: 'small' | 'medium' | 'large'
-  variant?: 'folderTree' | 'indent'
-  collectionIcon?: React.ReactNode | ((...args: any[]) => any)
-  collectionIconExpanded?: React.ReactNode | ((...args: any[]) => any)
-  itemIcon?: React.ReactNode | ((...args: any[]) => any)
-  getItemProps?: (...args: any[]) => any
-  getCollectionProps?: (...args: any[]) => any
-  onItemClick?: (...args: any[]) => any
-  onCollectionClick?: (...args: any[]) => any
-  onKeyDown?: (...args: any[]) => any
+  onKeyDown?: (e: React.KeyboardEvent, data: CollectionData) => void
+  // TODO it would be nice to use the same handler as in TreeBrowser
+  onItemClick?: (e: React.MouseEvent, data: CollectionData) => void
   numChildren?: number
-  level?: number
   position?: number
-  renderBeforeItems?: any // TODO: Children.oneOf([TreeNode])
-  renderAfterItems?: any // TODO: Children.oneOf([TreeNode])
-  containerRef?: (...args: any[]) => any
```

Comment:
Just a note here: there are some types that show up correctly on the docs page (eg TreeCollection's containerRef) because of the imports back and forth :/

Reply:
I could not get the text show up for this one.. I've still added some text for the IDE

Reply:
I think it is important info for the render functions that they have to be TreeNode components, because we cannot rely on the type either (we couldn't type this well yet). So I suggest using the fallback option to leave the descriptions for these props above their PropTypes, because react-docgen picks it up from there too.

Reply:
I think I have a solution which shows type information for the `containerRef, renderBeforeItems, renderAfterItems` functions for the `TreeCollection` component. The issue might be with the TS utility type `Omit<>`, but this is just the guess.
This is how I made it show type info:
Since the `Omit` is not working properly:
```ts
type CollectionProps = {
  collections?: CollectionProps[]
  items?: CollectionItem[]
  expanded?: boolean
  isCollectionFlattened?: boolean
} & Omit<Collection, 'collections' | 'items'>
```

What about reversing the `Collection` type and instead of Omitting stuff from it we would create 2 seperate types, like this:
```ts
type CollectionBase = {
  id: number | string
  name: string
  descriptor?: string
  /**
   * A function that returns a reference to the underlying HTML container
   * @param el The DOM HTMLElement
   */
  containerRef?: (el: HTMLElement | null) => void
  /**
   * children of type TreeNode
   */
  renderBeforeItems?: ReactElement // TODO: Children.oneOf([TreeNode])
  /**
   * children of type TreeNode
   */
  renderAfterItems?: ReactElement // TODO: Children.oneOf([TreeNode])
}
type Collection = CollectionBase & {
  items?: number[]
  collections?: (number | string)[]
}
```

And then, we can reuse the `CollectionBase` for the `CollectionProps`: 
```ts
type CollectionProps = {
  collections?: CollectionProps[]
  items?: CollectionItem[]
  expanded?: boolean
  isCollectionFlattened?: boolean
} & CollectionBase
```
Doing this way we have proper type info for the `Tree.Collection`:
<img width="942" alt="Screenshot 2022-01-20 at 10 32 41" src="https://user-images.githubusercontent.com/20640237/150313577-29b7ff28-1d93-42b6-bdc3-8e7fb81058c0.png">

Reply:
great, this worked :)